ambry-api/src/main/java/com.github.ambry/server/StoreManager.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.store.Store;
+import java.util.List;
+
+
+public interface StoreManager {
+
+  /**
+   * Add a new BlobStore with given {@link ReplicaId}.
+   * @param replica the {@link ReplicaId} of the {@link Store} which would be added.
+   * @return {@code true} if adding store was successful. {@code false} if not.
+   */
+  public boolean addBlobStore(ReplicaId replica);
+
+  /**
+   * Remove store from storage manager.
+   * @param id the {@link PartitionId} associated with store
+   * @return {@code true} if removal succeeds. {@code false} otherwise.
+   */
+  public boolean removeBlobStore(PartitionId id);
+
+  /**
+   * Start BlobStore with given {@link PartitionId} {@code id}.
+   * @param id the {@link PartitionId} of the {@link Store} which would be started.
+   */
+  public boolean startBlobStore(PartitionId id);
+
+  /**
+   * Shutdown BlobStore with given {@link PartitionId} {@code id}.
+   * @param id the {@link PartitionId} of the {@link Store} which would be shutdown.
+   */
+  public boolean shutdownBlobStore(PartitionId id);
+
+  /**
+   * @param id the {@link PartitionId} to find the store for.
+   * @return the {@link Store} corresponding to the given {@link PartitionId}, or {@code null} if no store was found for
+   *         that partition, or that store was not started.
+   */
+  public Store getStore(PartitionId id);
+
+  /**
+   * Set BlobStore Stopped state with given {@link PartitionId} {@code id}.
+   * @param partitionIds a list {@link PartitionId} of the {@link Store} whose stopped state should be set.
+   * @param markStop whether to mark BlobStore as stopped ({@code true}) or started.
+   * @return a list of {@link PartitionId} whose stopped state fails to be updated.
+   */
+  public List<PartitionId> setBlobStoreStoppedState(List<PartitionId> partitionIds, boolean markStop);
+
+  /**
+   * Check if a certain partition is available locally.
+   * @param partition the {@link PartitionId} to check.
+   * @param localReplica the {@link ReplicaId} of the localreplica of the partition.
+   * @return {@code true} if the partition is available. {@code false} if not.
+   */
+  public ServerErrorCode isPartitionAvailable(PartitionId partition, ReplicaId localReplica);//todo fix this
```

Comment:
Return value doesn't match javadoc, and isXXX method typically returns boolean or throws exception.